Permissioned DEX XLS.md

```diff
@@ -0,0 +1,467 @@
+
+An open offer does not need to include a `DomainID` field.
+
+### 5.2. Offer Invalidity
+
+An offer on the orderbook can become invalid if the domain specified by the `DomainID` field is deleted, and will no longer be filled. It will be automatically deleted when crossed, like an unfunded offer.
+
+The same will be true if the credential that allows the offer's owner to be a member of the domain expires or is deleted.
+
+## 6. Transaction: `OfferCreate`
+
+The `OfferCreate` transaction creates an offer on the CLOB DEX. This transaction type already exists on the XRPL, but is being extended as a part of this spec to also support permissioned DEX domains.
+
+### 6.1. Fields
+
+<details>
+<summary>
+
+As a reference, [here](https://xrpl.org/docs/references/protocol/transactions/types/offercreate) are the existing fields for the `OfferCreate` transaction.
+</summary>
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`Expiration`| |`number`|`UInt32`|Time after which the offer is no longer active, in seconds since the Ripple Epoch.|
+|`OfferSequence`| |`number`|`UInt32`|An offer to delete first, specified in the same way as `OfferCancel`.|
+|`TakerGets`|✔️|`string`|`Amount`|The amount and type of currency being sold.|
+|`TakerPays`|✔️|`string`|`Amount`|The amount and type of currency being bought.|
+</details>
+
+We propose these additions:
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`DomainID`| |`string`|`Hash256`|The domain that the offer must be a part of.|
+
+### 6.2. Failure Conditions
+
+The existing set of failure conditions for `OfferCreate` will continue to exist.
+
+There will also be the following in addition, if the `DomainID` field is included:
+* The domain doesn't exist.
+* The offer is not a valid domain offer.
+
+### 6.3. State Changes
+
+The existing set of state changes for `OfferCreate` will continue to exist.
+
+If the `DomainID` is included in the `OfferCreate` transaction, whenever an offer is crossed:
+* If the crossed offer doesn't include a `DomainID`, the offer will be checked to see if it is a valid domain offer. If it does, it will fill that offer as per the rules of the `OfferCreate`'s parameters. If it doesn't, the offer will be skipped.
+* If the crossed offer does include a `DomainID`, it must match the `DomainID` in the transaction (even if each would be a valid domain offer in the other domain).
+* If there is an offer placed on the ledger, it will include the `DomainID`.
+
+If the `DomainID` is not included in the `OfferCreate` transaction, whenever an offer with a `DomainID` is crossed:
+* The offer outlined in the `OfferCreate` transaction will be checked to see if it is a valid domain offer. If it does, it will fill that offer as per the rules of the `OfferCreate`'s parameters. If it doesn't, the offer will be skipped.
+
```

Comment:
How do we iterate over permissioned offers?

Reply:
Currently I assumed you would iterate over _all_ offers and check which ones are permissioned. I know this may have performance implications.